File: app/src/main/java/com/seungjun/randomox/activity/ReqMailActivity.java
package com.seungjun.randomox.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.seungjun.randomox.BaseActivity;
import com.seungjun.randomox.R;
import com.seungjun.randomox.adapter.ReqMailListAdapter;
import com.seungjun.randomox.db.LetterDBData;
import com.seungjun.randomox.db.LetterDBUtils;
import com.seungjun.randomox.view.LetterPopup;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class ReqMailActivity extends BaseActivity {

    private static final String TAG = "ReqMailActivity";

    private ReqMailListAdapter adapter;

    @BindView(R.id.list_letter)
    RecyclerView letterListView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_req_mail);

        ButterKnife.bind(this);

        LinearLayoutManager manager = new LinearLayoutManager(this);
        manager.setOrientation(LinearLayoutManager.VERTICAL);

        letterListView.setLayoutManager(manager);

        adapter = new ReqMailListAdapter(this);
        adapter.setLetterDBData(LetterDBUtils.getInstance(this).getAllData());
        adapter.setLetterItemClick(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                LetterDBData date = (LetterDBData) view.getTag();

                LetterDBUtils.getInstance(ReqMailActivity.this).updateLetterReadState(date.get_id(), "Y");
                adapter.setLetterDBData(LetterDBUtils.getInstance(ReqMailActivity.this).getAllData());
                adapter.notifyDataSetChanged();

                LetterPopup letterPopup = new LetterPopup(ReqMailActivity.this);
                letterPopup.setPopupText(date.getLetter_req_text());
                letterPopup.show();
            }
        });

        letterListView.setAdapter(adapter);
    }

    @Override
    protected void onResume() {
        super.onResume();

    }

    @OnClick(R.id.top_back)
    public void back(){
        finish();
    }

    @OnClick(R.id.send_mail)
    public void clickSendMail(){
        startActivity(new Intent(this, SendMailActivity.class));
    }
}

File: app/src/main/java/com/seungjun/randomox/activity/ReqMailActivity.kt
package com.seungjun.randomox.activity

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.seungjun.randomox.BaseActivity
import com.seungjun.randomox.R
import com.seungjun.randomox.adapter.ReqMailListAdapter
import com.seungjun.randomox.db.LetterDBData
import com.seungjun.randomox.view.LetterPopup
import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_req_mail.*
import kotlinx.android.synthetic.main.activity_sendmail.*
import kotlinx.android.synthetic.main.view_top_bar.*
import org.jetbrains.anko.startActivity

class ReqMailActivity : BaseActivity() {

    companion object {
        private val TAG = "ReqMailActivity"
    }

    private val adapter by lazy {
        ReqMailListAdapter(this).apply {
            baseDisPosable.add(
                    letterDao.getAllData()
                        .subscribeOn(Schedulers.io())
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe({
                            setLetterDBData(it)
                            notifyDataSetChanged()
                        })
            )

            setLetterItemClick(View.OnClickListener {
                val data = it.tag as LetterDBData

                if(data.letter_read == "N"){
                    data.letter_read = "Y"
                    baseDisPosable.add(
                            Completable.fromCallable { letterDao.changeReadState(data) }
                                    .subscribeOn(Schedulers.io())
                                    .subscribe({
                                        letterDao.getAllData()
                                                .subscribeOn(Schedulers.io())
                                                .observeOn(AndroidSchedulers.mainThread())
                                                .subscribe({
                                                    setLetterDBData(it)
                                                    notifyDataSetChanged()
                                                })
                                    })
                    )
                }

                LetterPopup(this@ReqMailActivity).apply {
                    setPopupText(data.letter_req_text)
                    show()
                }

            })
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_req_mail)

        list_letter.apply {
            layoutManager = LinearLayoutManager(this@ReqMailActivity).apply {
                orientation = LinearLayoutManager.VERTICAL
            }
            adapter = this@ReqMailActivity.adapter
        }

        top_back.setOnClickListener {
            finish()
        }

        send_mail.setOnClickListener {
            startActivity<SendMailActivity>()
        }

    }


}
